### PR TITLE
AI - Better shade cover

### PR DESCRIPTION
**Pull request by SafwatHalaby**
Changed the conditions of when the AI should remain in shade (doesn't fix the mushroom issue). Specifically, made the zombies light-sane even when idle and not chasing player.

The change is pretty trivial, it is very unlikely to break anything.

### CODE REVIEW

File: src/Mobs/Monster.cpp [hunk 1]
```diff
@@ -314,7 +314,15 @@ void cMonster::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 	HandleDaylightBurning(*Chunk, WouldBurnAt(GetPosition(), *Chunk));
 	if (TickPathFinding(*Chunk))
 	{
-		if (m_BurnsInDaylight && WouldBurnAt(m_NextWayPointPosition, *Chunk->GetNeighborChunk(FloorC(m_NextWayPointPosition.x), FloorC(m_NextWayPointPosition.z))) && !IsOnFire() && (m_TicksSinceLastDamaged == 100))
+		/* If I burn in daylight, and I won't burn where I'm standing, and I'll burn in my next position, and at least one of those is true:
+		1. I am idle
+		2. I was not hurt by a player recently.
+		Then STOP. */
+		if (
+			m_BurnsInDaylight && ((m_TicksSinceLastDamaged == 100) || (m_EMState == IDLE)) &&
```
Comment (bearbin) [inline src/Mobs/Monster.cpp]: Should this be >= 100 rather than == 100?

Comment (SafwatHalaby) [inline src/Mobs/Monster.cpp]: m_TicksSinceLastDamaged maxes at100 and doesn't increase beyond that. But `>=` is better, just in case this is ever changed.

Comment (SafwatHalaby) [inline src/Mobs/Monster.cpp]: Done.

**Comment by SafwatHalaby** [comment]
Merge?

**Pull request merged**